pallets/pool-system/src/lib.rs

```diff
@@ -260,6 +261,8 @@ pub mod pallet {
 
 		type CurrencyId: Parameter + Copy + MaxEncodedLen;
 
+		type ExtChange: Parameter + Member + MaxEncodedLen + TypeInfo + Into<ExtPoolChange>;
```

Comment:
Why `ExtChange`? I think we should use this for both `loan` and `pool` changes, right, so then only some of them are external? 

We should also add a TODO (can be done later of course) to migrate https://github.com/centrifuge/centrifuge-chain/blob/main/pallets/pool-system/src/impls.rs#L199 to use this `ChangeGuard` implementation

Reply:
Make sense!

Reply:
Issue to track the migration: https://github.com/centrifuge/centrifuge-chain/issues/1397